src/couch/src/couch_db.erl

```diff
@@ -1338,6 +1338,15 @@ update_docs(Db, Docs0, Options, ?REPLICATED_CHANGES) ->
     ),
     {ok, DocErrors};
 update_docs(Db, Docs0, Options, ?INTERACTIVE_EDIT) ->
+    BlockInteractiveDatabaseWrites = couch_disk_monitor:block_interactive_database_writes(),
+    if
+        BlockInteractiveDatabaseWrites ->
+            {ok, [{insufficient_storage, <<"database_dir is too full">>}|| _ <- Docs0]};
+        true ->
+            update_docs_interactive(Db, Docs0, Options)
+    end.
+
+update_docs_interactive(Db, Docs0, Options) ->
```

Comment:
We should have some tests to assert what happens with single doc puts and bulk_docs. Could probably some existing fabric doc update tests or add some new ones (could be in chttpd too, if they are at the HTTP api level) then either mock `couch_disk_monitor:block_interactive_database_writes()` or maybe configure a very low limit and check that we get an expected result.